GetIntoTeachingApi/Services/Store.cs

```diff
@@ -113,58 +113,64 @@ private void SyncTeachingEvents(ICrmService crm)
+            UpsertAndRemoveOrphanedModels(teachingEvents, _dbContext.TeachingEvents);
 
             _dbContext.SaveChanges();
         }
 
         private void SyncPrivacyPolicies(ICrmService crm)
         {
             var policies = crm.GetPrivacyPolicies().ToList();
-            UpsertModels(policies, _dbContext.PrivacyPolicies);
-            _dbContext.SaveChanges();
+            UpsertAndRemoveOrphanedModels(policies, _dbContext.PrivacyPolicies);
         }
 
         private void SyncTypeEntities(ICrmService crm)
         {
-            UpsertTypes(crm.GetLookupItems("dfe_country"));
-            UpsertTypes(crm.GetLookupItems("dfe_teachingsubjectlist"));
-            UpsertTypes(crm.GetPickListItems("contact", "dfe_ittyear"));
-            UpsertTypes(crm.GetPickListItems("contact", "dfe_preferrededucationphase01"));
-            UpsertTypes(crm.GetPickListItems("contact", "dfe_isinuk"));
-            UpsertTypes(crm.GetPickListItems("contact", "dfe_channelcreation"));
-            UpsertTypes(crm.GetPickListItems("dfe_qualification", "dfe_degreestatus"));
-            UpsertTypes(crm.GetPickListItems("dfe_qualification", "dfe_category"));
-            UpsertTypes(crm.GetPickListItems("dfe_qualification", "dfe_type"));
-            UpsertTypes(crm.GetPickListItems("dfe_candidatepastteachingposition", "dfe_educationphase"));
-            UpsertTypes(crm.GetPickListItems("msevtmgt_event", "dfe_event_type"));
-            UpsertTypes(crm.GetPickListItems("phonecall", "dfe_channelcreation"));
-            _dbContext.SaveChanges();
+            UpsertAndRemoveOrphanedTypes(crm.GetLookupItems("dfe_country"));
+            UpsertAndRemoveOrphanedTypes(crm.GetLookupItems("dfe_teachingsubjectlist"));
+            UpsertAndRemoveOrphanedTypes(crm.GetPickListItems("contact", "dfe_ittyear"));
+            UpsertAndRemoveOrphanedTypes(crm.GetPickListItems("contact", "dfe_preferrededucationphase01"));
+            UpsertAndRemoveOrphanedTypes(crm.GetPickListItems("contact", "dfe_isinuk"));
+            UpsertAndRemoveOrphanedTypes(crm.GetPickListItems("contact", "dfe_channelcreation"));
+            UpsertAndRemoveOrphanedTypes(crm.GetPickListItems("dfe_qualification", "dfe_degreestatus"));
+            UpsertAndRemoveOrphanedTypes(crm.GetPickListItems("dfe_qualification", "dfe_category"));
+            UpsertAndRemoveOrphanedTypes(crm.GetPickListItems("dfe_qualification", "dfe_type"));
+            UpsertAndRemoveOrphanedTypes(crm.GetPickListItems("dfe_candidatepastteachingposition", "dfe_educationphase"));
+            UpsertAndRemoveOrphanedTypes(crm.GetPickListItems("msevtmgt_event", "dfe_event_type"));
+            UpsertAndRemoveOrphanedTypes(crm.GetPickListItems("phonecall", "dfe_channelcreation"));
         }
 
-        private void UpsertModels<T>(IEnumerable<T> models, IQueryable<T> dbSet) where T : BaseModel
+        private void UpsertAndRemoveOrphanedModels<T>(IEnumerable<T> models, IQueryable<T> dbSet) where T : BaseModel

```

Comment:
Really minor comment on naming - is UpsertAndRemoveOrphaned a long way of saying Synchronize? That said it is more explicit though?

Reply:
Ah! thank you - I couldn't see the woods for the trees on that one 😁